### ISSUE TITLE
`epinio namespace list` too slow

### ISSUE THREAD

**Issue by jimmykarily** [epinio/epinio#1353]

**Comment by richard-cox** [comment]
Just bringing across a comment from slack

`There were a lot of requests UI side for additional information in these types of api requests. We should be careful to avoid regressions/breakages that side. Pretty sure though there's some compromises or changes UI side can make`

The apps and configurations within the namespace request were added to add additional value to the namespace list, though at the moment only the count for each type is shown (originally this was going to list those resources as well).

Thinking about it, the info regarding resource counts per namespace could be pulled out into a single request, perhaps `/counts` or `/stats`? That could then include other things like total number of services, routes, etc and could be the basis of a landing page containing general cluster info (as well as providing the counts for the namespace page).

**Comment by enrichman** [comment]
Unfortunately it's not a problem regarding the number of information requested, but about the number of requests. And since we are not using any persistence layer, but we are only relying about the k8s API then I guess that to have a `/counts` we still have to do the same queries that we are doing right now.

**Comment by andreas-kupries** [comment]
Reading the latest comments it seems to me that your a talking across each other.
I suspect that the reason for that is the simple use `request`. That word is ambiguous as-is because we more than one kind of request here.

1. We have requests from the UI and CLI to the epinio server. 
2. And then there are the requests made by the epinio server to the kube cluster.

Let us call the first `epinio API requests`. And the second are `k8s requests`.

It is my reading that Richard is concerned that changes to the epinio server would change the API of the server and the request UI would have to make.

My proposals, currently only shown in slack thread would not change the epinio API. It would only change the k8s requests made by the server. The server would then reassemble the same information into the existing responses.

My proposal was to use kube `List` commands with label selectors to retrieve all necessary information about namespaces and the applications and configurations therein in a fixed number of few calls (less than 10 ?) instead of making namespace-specific calls and/or even per-app calls.

**Comment by enrichman** [comment]
Yes exactly, I would keep the Epinio API the same, also if we don't need all the data of the resources, and change how we make requests to the k8s APIs. :+1:

**Comment by richard-cox** [comment]
Just tracking here, it is possible to get the number of resource via the kube API 

https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks

`/api/v1/pods?limit=1`and the response should contain a `metadata.remainingItemCount`. That would need to happen per namespace, but it should be pretty quick

**Comment by richard-cox** [comment]
I thought selectors were distinct from namespaces/params? Might need some investigation
